### PR TITLE
Importer schedule code overrides, program_type logic updates

### PR DESCRIPTION
**Pull request by cjg89**
- Added ability to filter a degree's determined schedule code in themes/other plugins (`ucf_tuition_fees_get_schedule_code` hook)
- Updated logic for determining a program's program_type to use a parent program_type term if no child is found

### CODE REVIEW

**Review by rjucf — APPROVED** [review]

File: importers/tuition-fees-importer.php [hunk 1]
```diff
@@ -233,15 +233,24 @@ private function update_degrees() {
...
+
 			$plan_code    = get_post_meta( $degree->ID, UCF_Tuition_Fees_Config::get_option_or_default( 'degree_plan_code_name' ), true );
 			$subplan_code = get_post_meta( $degree->ID, UCF_Tuition_Fees_Config::get_option_or_default( 'degree_subplan_code_name' ), true );
 			$is_online    = filter_var( get_post_meta( $degree->ID, UCF_Tuition_Fees_Config::get_option_or_default( 'degree_online_meta_field' ), true ), FILTER_VALIDATE_BOOLEAN );
 
 			// If no program type, skip it
 			if ( ! $program_type ) { $this->skipped_total++; continue; }
 
-			$schedule_code = $this->get_schedule_code( $program_type->name, $plan_code, $subplan_code, $is_online );
+			$schedule_code = $this->get_schedule_code( $degree, $program_type->name, $plan_code, $subplan_code, $is_online );
```
Comment (jmbarne3) [inline importers/tuition-fees-importer.php:253]: The `$program_type->name` is going to throw a warning if `$program_type` is null (which is an option based on line 243). We can avoid the warning in the log by putting the name in a variable, or setting it to null if `$program_type` is null.

**Review by cadie — APPROVED** [review]

File: importers/tuition-fees-importer.php [hunk 2]
```diff
@@ -262,40 +271,48 @@ private function update_degrees() {
 		}
 	}
 
-	private function get_schedule_code( $program_type, $plan_code, $subplan_code, $is_online ) {
+	private function get_schedule_code( $degree, $program_type, $plan_code, $subplan_code, $is_online ) {
```
Comment (jmbarne3) [inline importers/tuition-fees-importer.php:274]: It looks like most of the logic in this script is dependent on `$program_type` having a value. We should do a null check early on in the function and short circuit the function if it is.

**Review by jmbarne3 — CHANGES_REQUESTED** [review]

File: importers/tuition-fees-importer.php (hunk 2 above)
Comment (cjg89) [reply importers/tuition-fees-importer.php:274]: re. both comments:  We do a short circuit on line 251 and prevent any null `$program_type`'s from getting passed into `get_schedule_code()` to begin with.  We could add an extra `$program_type` check in `get_schedule_code()` for extra hardening if we wanted to, but in its current state, we shouldn't ever see any warnings thrown due to `$program_type` being null.

Comment (jmbarne3) [reply importers/tuition-fees-importer.php:274]: Ok. As long as `get_schedule_code` stays private, and isn't used anywhere else in the class, it should be fine. If we ever think we might want to make it public, adding a null check will make it a little safer for use.

**Review by jmbarne3 — APPROVED** [review]

**Pull request merged**